.travis.yml

```diff
@@ -2,4 +2,30 @@ language: java
 sudo: false
```

Comment:
(Dropping this note here to allow for threading (I dislike that GitHub doesn't allow threaded conversations on the base conversation page.))

Ideally I'd like to integrate this logic with the maven release plugin, to avoid needing to futz with the versions manually and potentially avoid having to push manually to a separate release branch.  Do you see that as being a fairly straightforward extension / follow-up to your work here?

On a related note, I'm unclear on exactly how the uploading of the release is "gated" so that it will only happen when we create a release?  (I'm *guessing* however that this is some implicit behavior from the travis handling of the "deploy provider" being "releases" below.)   Also, we do wanna keep having travis builds for normal pushes, I just wanna make sure this change isn't preventing that.